cumulusci/cli/tests/test_run_task.py

```diff
@@ -0,0 +1,156 @@
+
+color_opts = {"options": {"color": {}}}
+multiple_opts = {"options": {"foo": {}, "bar": {}, "baz": {}}}
+
+test_tasks = {
+    "dummy-task": {"class_path": "cumulusci.cli.tests.utils.DummyTask"},
+    "dummy-derived-task": {
+        "class_path": "cumulusci.cli.tests.test_run_task.DummyDerivedTask"
+    },
+}
+
+
+@pytest.fixture
+def ctx():
+    ctx = Mock()
+    ctx.task_class = DummyTask
+    return ctx
+
+
+@pytest.fixture
+def runtime():
+    runtime = CliRuntime(load_keychain=False)
+    runtime.project_config.config["tasks"] = {**test_tasks}
+
+    runtime.keychain = Mock()
+    runtime.keychain.get_default_org.return_value = (None, None)
+
+    with patch("cumulusci.cli.cci.RUNTIME", runtime):
+        yield runtime
+
+
+def test_task_run(runtime, ctx):
+    DummyTask._run_task = Mock()
+    multi_cmd = cci.RunTaskCommand()
+    cmd = multi_cmd.get_command(ctx, "dummy-task")
+
+    run_click_command(cmd, "dummy-task", color="blue", runtime=runtime)
+
+    DummyTask._run_task.assert_called_once()
+
+
+def test_task_run__debug_before(runtime, ctx):
+    DummyTask._run_task = Mock()
+    multi_cmd = cci.RunTaskCommand()
+    set_trace = Mock(side_effect=SetTrace)
+
+    with patch("pdb.set_trace", set_trace):
+        with pytest.raises(SetTrace):
+            cmd = multi_cmd.get_command(ctx, "dummy-task")
+            run_click_command(
+                cmd,
+                "dummy_task",
+                color="blue",
+                debug_before=True,
+                debug_after=False,
+                runtime=runtime,
+            )
+
+
+def test_task_run__debug_after(runtime, ctx):
+    DummyTask._run_task = Mock()
+    multi_cmd = cci.RunTaskCommand()
+
+    set_trace = Mock(side_effect=SetTrace)
+    with patch("pdb.set_trace", set_trace):
+        with pytest.raises(SetTrace):
+            cmd = multi_cmd.get_command(ctx, "dummy-task")
+            run_click_command(
+                cmd,
+                "dummy-task",
+                color="blue",
+                debug_before=False,
+                debug_after=True,
+                runtime=runtime,
+            )
+
+
+def test_task_run__list_commands(runtime, ctx):
+    multi_cmd = cci.RunTaskCommand()
+    commands = multi_cmd.list_commands(ctx)
+    assert commands == ["dummy-derived-task", "dummy-task"]
+
+
+def test_format_help__proj_conf_exists(runtime):
+    with patch("cumulusci.cli.cci.click.echo") as echo:
+        runtime.universal_config = Mock()
+        RunTaskCommand().format_help(Mock(), Mock())
+        assert 4 == echo.call_count
+        assert 0 == len(runtime.universal_config.method_calls)
+
+
+def test_format_help__proj_conf_does_not_exist(runtime):
+    with patch("cumulusci.cli.cci.click.echo") as echo:
+        runtime.universal_config = runtime.project_config
+        runtime.project_config = None
+        RunTaskCommand().format_help(Mock(), Mock())
+        assert 4 == echo.call_count
```

Comment:
Probably don't need these 2 different tests for format_help now that get_available_tasks takes care of the difference between having a project config or not?